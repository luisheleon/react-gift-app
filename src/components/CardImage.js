import * as React from "react";
import Card from "@mui/material/Card";
import CardContent from "@mui/material/CardContent";
import CardMedia from "@mui/material/CardMedia";
import Typography from "@mui/material/Typography";
import "animate.css";

export default function MediaCard({ id, title, url }) {
  return (
    <Card sx={{ maxWidth: 345 }} className="animate__animated animate__fadeIn">
      <CardMedia component="img" height="140" image={url} alt={id + title} />
      <CardContent className="cardTitle">
        <Typography gutterBottom variant="caption" component="div">
          {title}
        </Typography>
      </CardContent>
    </Card>
  );
}
