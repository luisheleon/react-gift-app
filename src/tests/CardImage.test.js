import CardImage from "../components/CardImage";
import { mount } from "enzyme";
import CardMedia from "@mui/material/CardMedia";
import Typography from "@mui/material/Typography";
import { shallow } from "enzyme";

describe("test <CardImage />", () => {
  const props = {
    title: "Un titulo",
    id: "asdfasdf87238974",
    url: "http://www.google.com",
  };

  let wrapper, wrapperMount;
  beforeEach(() => {
    wrapperMount = mount(<CardImage {...props} />);
    wrapper = shallow(<CardImage {...props} />);
  });
  test("should render component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should create a cardMedia and Typography components whit props", () => {
    const img = wrapperMount.find(CardMedia);
    const typo = wrapperMount.find(Typography);
    const imgProps = img.props();
    const typoProps = typo.props();
    expect(img).toHaveLength(1);
    expect(imgProps.image).toBe(props.url);
    expect(imgProps.alt).toBe(props.id + props.title);
    expect(typo).toHaveLength(1);
    expect(typoProps.children).toBe(props.title);
  });
});
