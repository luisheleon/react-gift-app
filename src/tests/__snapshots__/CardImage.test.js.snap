// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <CardImage /> should render component 1`] = `
<ForwardRef(Card)
  className="animate__animated animate__fadeIn"
  sx={
    Object {
      "maxWidth": 345,
    }
  }
>
  <ForwardRef(CardMedia)
    alt="asdfasdf87238974Un titulo"
    component="img"
    height="140"
    image="http://www.google.com"
  />
  <ForwardRef(CardContent)
    className="cardTitle"
  >
    <ForwardRef(Typography)
      component="div"
      gutterBottom={true}
      variant="caption"
    >
      Un titulo
    </ForwardRef(Typography)>
  </ForwardRef(CardContent)>
</ForwardRef(Card)>
`;
